File: js/euler65.js
function euler65(){
    var res = 0;
    var num = 2n, prev = 1n;
    var k = 1n;
    for (let i = 1; i < 100; i++){
        // 1, 2, 1, 1, 4, 1, 1, 6, 1, ..., 1, 2k, 1, ...
        let r = (i % 3 == 2) ? 2n * k++ : 1n;
        newNum = num * r + prev;
        prev = num;
        num = newNum;
    }
    for (let c of num.toString()){
        res += Number(c);
    }
    return res;
}

console.log(euler65());

File: go/euler65.go
package main
import (
    "fmt"
    "math/big"
)

func getDigitSum(n *big.Int) int {
    temp := new(big.Int).Set(n)
    zero := big.NewInt(0)
    mod := big.NewInt(0)
    ten := big.NewInt(10)
    sum := 0
    for temp.Cmp(zero) != 0 {
        temp.DivMod(temp, ten, mod)
        sum += int(mod.Int64())
    }
    return sum
}

func main() {
    k := 1
    prevNum := big.NewInt(1)
    num := big.NewInt(2)
    for n := 1; n < 100; n++ {
        r := int64(1)
        if n % 3 == 2 {
            r = int64(2 * k)
            k++
        }
        temp := new(big.Int)
        newNum := temp.Add(temp.Mul(big.NewInt(r), num), prevNum)
        prevNum = num
        num = newNum
    }
    res := getDigitSum(num)
    fmt.Println(res)
}
